test/extended/baremetal/hosts.go

```diff
@@ -33,6 +34,29 @@ func baremetalClient(dc dynamic.Interface) dynamic.ResourceInterface {
 	return baremetalClient.Namespace("openshift-machine-api")
 }
 
+type FieldGetterFunc func(obj map[string]interface{}, fields ...string) (interface{}, bool, error)
+
+func expectField(host unstructured.Unstructured, nestedField string, fieldGetter FieldGetterFunc) o.Assertion {
+	fields := strings.Split(nestedField, ".")
+
+	value, found, err := fieldGetter(host.Object, fields...)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(found).To(o.BeTrue(), "baremetalhost field not found")
```

Comment:
instead "baremetalhost", nestedField can be used in here.

Reply:
Good point, thanks